ext/SerializationExt.jl

```diff
@@ -0,0 +1,82 @@
+module SerializationExt
+using LRUCache
+using Serialization
+
+# Serialization of large LRUs causes a stack overflow error, so we 
+# create a custom serializer that represents LinkedNodes as Ints
+function Serialization.serialize(s::AbstractSerializer, lru::LRU{K, V}) where {K, V}
+    # Create a mapping from memory address to id
+    node_map = IdDict{LRUCache.LinkedNode{K}, Int}()
```

Comment:
I think you can add a size hint here:
```suggestion
    node_map = IdDict{LRUCache.LinkedNode{K}, Int}()
    sizehint!(node_map, length(lru))
```

Reply:
Brilliant! Never heard about `sizehint!` before.